rosbag2_cpp/include/rosbag2_cpp/reader_interfaces/base_reader_interface.hpp

```diff
@@ -51,6 +51,10 @@ class ROSBAG2_CPP_PUBLIC BaseReaderInterface
 
   virtual const rosbag2_storage::BagMetadata & get_metadata() const = 0;
 
+  virtual void setMetadataAllow0MessageCount(__attribute__((unused)) bool enable)
+  {
+  }
+
```

Comment:
i am not sure why we need this into this class... instead we can address the issue in sqlite3 plugin implementation?

Reply:
The problem is that the sequential reader uses the base interface and not directly the sqlite implementation of it. So it is necessary to complete the interface by this function, then override it inside sqlite implementation. But maybe there is others solutions.